charts/opentelemetry-demo/templates/_objects.tpl

```diff
@@ -19,16 +19,17 @@ spec:
         {{- toYaml .podAnnotations | nindent 8 }}
       {{- end }}
     spec:
-      {{- if .imageConfig.pullSecrets }}
+      {{- if or .defaultValues.image.pullSecrets .imageOverride.pullSecrets }}
       imagePullSecrets:
-        {{- toYaml .imageConfig.pullSecrets | nindent 8}}
+        {{- .imageOverride.pullSecrets | default .defaultValues.image.pullSecrets | toYaml | nindent 8}}
       {{- end }}
       {{- with .serviceAccountName }}
       serviceAccountName: {{ .serviceAccountName}}
       {{- end }}
       containers:
         - name: {{ .name }}
-          image: {{ .image | default (printf "%s:v%s-%s" .imageConfig.repository .Chart.AppVersion (.name | replace "-" "" | lower)) }}
+          image: '{{ .imageOverride.repository | default .defaultValues.image.repository }}:{{ .imageOverride.tag | default (printf "v%s-%s" (default .Chart.AppVersion .defaultValues.image.tag) (.name | replace "-" "" | lower)) }}'
```

Comment:
Why `replace "-" ""` is needed? Do we have any names with `-`?

Reply:
Ya, because we pipe the values.yaml name through kebabcase: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/edb0c9316a43c687395cb6ae32c3ff69dac8aa32/charts/opentelemetry-demo/templates/component.yaml#L3.  This is great for the name, but the tag doesn't use it: https://hub.docker.com/r/otel/demo/tags

Reply:
We could get rid of the `| lower` tho